_ml-commons-plugin/api/model-apis/register-model.md

```diff
@@ -248,6 +249,68 @@ OpenSearch responds with the `task_id` and task `status`.
 }
 ```
 
+### The `guardrails` parameter
+
+Guardrails act like safety measures for large language models (LLMs). They provide a set of rules and boundaries that control how an LLM behaves and what kind of outputs it generates. 
+
+To register an externally hosted model with guardrails, provide the `guardrails` parameter, which supports the following fields. All fields are optional.
+
+Field | Data type | Description
+:---  | :--- | :---
+`input_guardrail`| Object |  The guardrail for the model input. |
+`output_guardrail`| Object |  The guardrail for the model output. |
+`stop_words`| Object | The list of indexes containing stopwords used for the model input/output validation. If the model prompt/response contains a stopword contained in any of the indexes, the predict request on this model is rejected. |
+`index_name`| Object | The name of the index storing the stopwords. |
+`source_fields`| Object | The name of the field storing the stopwords. |
+`regex`| Object |  A regular expression used for input/output validation. If the model prompt/response matches the regular expression, the predict request on this model is rejected. |
+
+#### Example request: Externally hosted model with guardrails
+
+```json
+POST /_plugins/_ml/models/_register
+{
+  "name": "openAI-gpt-3.5-turbo",
+  "function_name": "remote",
+  "model_group_id": "1jriBYsBq7EKuKzZX131",
+  "description": "test model",
+  "connector_id": "a1eMb4kBJ1eYAeTMAljY",
+  "guardrails": {
+    "input_guardrail": {
```

Comment:
Our testing has shown that a "type" field is required here or it fails parsing.  @jngz-es @ylwu-amzn

Reply:
Thank you, @dbwiddis! I have added the `type` parameter in this commit https://github.com/opensearch-project/documentation-website/pull/6750/commits/2b13532d4d1a957663fd8e3e4037d1fb254fc670